libsupport/src/ProgressTracer.cpp

```diff
@@ -24,10 +24,11 @@ katana::ProgressTracer::StartActiveSpan(
 void
 katana::ProgressTracer::FinishActiveSpan() {
   if (active_span_ != nullptr) {
-    if (!active_span_->IsFinished()) {
-      active_span_->Finish();
+    auto old_active_span = active_span_;
+    active_span_ = old_active_span->GetParentSpan();
+    if (!old_active_span->IsFinished()) {
+      old_active_span->Finish();
```

Comment:
I don't have a lot of context, but is it possible to be in a much deeper span-nest, so this should really be a `while old_active_span->GetParentSpan()` construct?

Reply:
good catch, added some logic to handle this recursively in case someone is ending spans early